Frameworks/Foundation/NSIndexPath.mm

```diff
@@ -90,114 +112,95 @@ - (BOOL)isEqual:(id)other {
 /**
  @Status Interoperable
 */
-- (void)dealloc {
-    if (_indexes) {
-        IwFree(_indexes);
-    }
-    [super dealloc];
-}
-
-/**
- @Status Caveat
- @Notes position must be valid
-*/
-- (unsigned)indexAtPosition:(unsigned)position {
-    if (position >= _length) {
-        assert(0);
-        //[NSException raise:NSInvalidArgumentException format:@"Unable to remove index from zero length path."];
-        return 0;
-    }
-
-    return _indexes[position];
+- (NSUInteger)indexAtPosition:(NSUInteger)position {
+    return position < _indexes.size() ? _indexes[position] : NSNotFound;
 }
 
 /**
  @Status Interoperable
 */
 - (void)getIndexes:(NSUInteger*)indexes {
-    for (unsigned int i = 0; i < _length; i++) {
-        indexes[i] = _indexes[i];
-    }
+    std::copy(_indexes.cbegin(), _indexes.cend(), indexes);
 }
 
 /**
  @Status Interoperable
 */
 - (id)copyWithZone:(NSZone*)zone {
-    return [[[self class] alloc] initWithIndexes:_indexes length:_length];
+    return [[[self class] alloc] initWithIndexes:_indexes.data() length:_indexes.size()];
 }
 
 /**
  @Status Interoperable
 */
 - (NSIndexPath*)indexPathByAddingIndex:(NSUInteger)newIndex {
-    id ret = [[self class] alloc];
-
-    NSUInteger* indexCopy = (NSUInteger*)IwMalloc((_length + 1) * sizeof(DWORD));
-    memcpy(indexCopy, _indexes, _length * sizeof(DWORD));
-    indexCopy[_length] = newIndex;
-
-    [ret initWithIndexes:indexCopy length:_length + 1];
-
-    IwFree(indexCopy);
+    std::vector<NSUInteger> indexes = _indexes;
+    indexes.emplace_back(newIndex);
```

Comment:
the extra copy here is unfortunate; how about just alloc'ing an empty NSIndex path, then std::move(indexes) into the new object's _indexes vector?